### ISSUE TITLE
PFN training on tf.data.Dataset objects

### ISSUE THREAD

**Issue by russbate42** [pkomiske/EnergyFlow#23]
Hello,
My pfn network seems to have some kind of incompatibility with training on Dataset objects. Using numpy arrays the training works fine. However, I'm trying to construct a data pipeline in order to handle larger data sets as well as use keras tools for multi GPU training. I'm constructing this object in the following way:

deepSets = tf.data.Dataset.from_tensor_slices((X, Y))
ds_train = deepSets.skip(val+test)
ds_test = deepSets.take(test+val)
ds_val = ds_test.skip(test)
ds_test = ds_test.take(test)

pfn = PFN(input_dim=5, Phi_sizes=Phi_sizes, F_sizes=F_sizes, 
          output_act=output_act, output_dim=output_dim, loss=loss,
          optimizer=optimizer, metrics=[])

history = pfn.fit(ds_train,
        epochs=num_epoch,
        batch_size=batch_size,
        validation_data=ds_val,
        verbose=1)

Now I get the following error:
Epoch 1/1500
WARNING:tensorflow:Model was constructed with shape (None, None, 5) for input KerasTensor(type_spec=TensorSpec(shape=(None, None, 5), dtype=tf.float32, name='input'), name='input', description="created by layer 'input'"), but it was called on an input with incompatible shape (1001, 5)

What is the correct way to make a PFN model compatible with a tf.data.Dataset object?

**Comment by russbate42** [comment]
Solved, we can delete this.
